From 24b2db03708e6225b9224f14a632e51f8782c9ec Mon Sep 17 00:00:00 2001
From: Teguh Sobirin <teguh@sobir.in>
Date: Wed, 29 Jun 2022 05:07:06 +0700
Subject: [PATCH] iio: adc: qcom-spmi-adc5: add more channels needed by AYN
 Odin gamepad

ADC5_GPIO1, ADC5_GPIO2 ADC5_GPIO3, ADC5_GPIO4, ADC5_GPIO5, ADC5_GPIO6_DIV3
---
 drivers/iio/adc/qcom-spmi-adc5.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/iio/adc/qcom-spmi-adc5.c b/drivers/iio/adc/qcom-spmi-adc5.c
index af3c2f659f5e..998167257093 100644
--- a/drivers/iio/adc/qcom-spmi-adc5.c
+++ b/drivers/iio/adc/qcom-spmi-adc5.c
@@ -623,6 +623,18 @@ static const struct adc5_channels adc5_chans_rev2[ADC5_MAX_CHANNEL] = {
 					SCALE_HW_CALIB_THERM_100K_PULLUP)
 	[ADC5_XO_THERM_100K_PU]	= ADC5_CHAN_TEMP("xo_therm_100k_pu", 0,
 					SCALE_HW_CALIB_THERM_100K_PULLUP)
+	[ADC5_GPIO1]		= ADC5_CHAN_VOLT("gpio1", 0,
+					SCALE_HW_CALIB_DEFAULT)
+	[ADC5_GPIO2]		= ADC5_CHAN_VOLT("gpio2", 0,
+					SCALE_HW_CALIB_DEFAULT)
+	[ADC5_GPIO3]		= ADC5_CHAN_VOLT("gpio3", 0,
+				SCALE_HW_CALIB_DEFAULT)
+	[ADC5_GPIO4]		= ADC5_CHAN_VOLT("gpio4", 0,
+				SCALE_HW_CALIB_DEFAULT)
+	[ADC5_GPIO5]		= ADC5_CHAN_VOLT("gpio5", 0,
+					SCALE_HW_CALIB_DEFAULT)
+	[ADC5_GPIO6_DIV3]	= ADC5_CHAN_VOLT("gpio6", 0,
+				SCALE_HW_CALIB_DEFAULT)
 };
 
 static int adc5_get_fw_channel_data(struct adc5_chip *adc,
-- 
GitLab

